java/org/apache/coyote/Response.java

```diff
@@ -61,6 +62,11 @@
      */
     private static final Locale DEFAULT_LOCALE = Locale.getDefault();
 
+    /**
+     * Helper to log the invalid HTTP/2.0 header error only once per instance
+     */
+    private static AtomicBoolean invalidHeaderWarningEmitted = new AtomicBoolean(false);
```

Comment:
`per instance` of what ? Because you use `static` here, so it is on class loader level.

Reply:
@martin-g It's a langage abuse. Per instance of running tomcat, sorry. Anyhow, @rmaucher suggestion to just throw the exception is simpler, so the AtomicBoolean is deprecated.